itou/antivirus/management/commands/scan_s3_files.py

```diff
@@ -45,12 +33,14 @@ def s3_obj_filter_func(obj_summary):
             region_name=settings.S3_STORAGE_BUCKET_REGION,
             config=Config(signature_version="s3v4"),
         )
-        paginator = self.client.get_paginator("list_objects_v2")
-        page_iterator = paginator.paginate(Bucket=options["bucket"])
-        scanned = 0
-        while True:
-            with tempfile.TemporaryDirectory() as workdir:
-                filepath_s3key, has_next = self.download_files(page_iterator, s3_obj_filter_func, workdir)
+        files = File.objects.exclude(
+            Exists(Scan.objects.filter(file_id=OuterRef("pk"), clamav_completed_at__gt=now - relativedelta(month=1)))
+        )[: self.BATCH_SIZE]
```

Comment:
Je me méfie des subqueries, peut-être:
```suggestion
        files = File.objects.select_related('scan').exclude(scan__clamav_completed_at__gt=now - relativedelta(month=1)))).order_by('scan__clamav_completed_at').asc(nulls_first=True)[: self.BATCH_SIZE]
```
?

Reply:
Pas besoin du `select_related("scan")`, mais c’est effectivement mieux.